ONSdigital/census-rm-deploy
# auto scale performance tests infrastructure

# Motivation and Context

There is no need for the performance tests infrastructure to be running 24/7, so tasks have been created to scale down certain parts of the infrastructure.
Following a successful performance test, the pipeline calls tasks located in the performance test repo to do the following:

- delete DB read replica (DB won't stop if a read replica is present)
- stop DB
- delete cluster node pools (not deleting whole cluster)

When the performance tests are triggered, a task runs to start the DB, terraform will then take care of adding the DB read replica and relevant node pools.

The pipeline has been simplified so there is now one route through, take a look at the RM concourse sandbox at the pipeline called hughs-test to see how it looks.

# What has changed



Performance test pipeline call tasks (located in census-rm-performance-tests repo) to scale up and down the infrastructure.

# How to test?



Validate the pipeline, check it alongside the PR in census-rm-performance-tests on the branch with the same name as this one:
https://github.com/ONSdigital/census-rm-performance-tests/pull/14

# Links



https://trello.com/c/Noi86T6D/268-scale-performance-environment-automatically-8

# Screenshots (if appropriate):
